ONLYOFFICE/build_tools
# Merge branch release/v7.5.0 into master

- init commit
- [build] license checker is ready (tested only for sdkjs)
- [build] small fix license checker
- [build] license checker added config for repos
- [build] fix license_checker deleted unused ignore
- [build] license checker updated config options Added ignoreListDirName
- [build] license checker fix docs
- [build] updated license checker
- fix license checker
- license_checker readme added
- fix
- [build] license_checker small fix
- [develop] Add plugins deploy for developer version
- [build] added checks for plugins
- [build] fix prev commit
- [build] fix plugins
- [build] add marketplace plugin at ignore
- [build] fix
- [license] Update config; Fix bug with bom, leading space; and minor changes
- [license] Update address in Copyright
- Revert "[develop] Add plugins deploy for developer version"
- Revert "[build] added checks for plugins"
- [build] small fix
- [build] fix deleted semicolons
- [build] revert base.py
- [build] revert base.py
- test doc
- [build] changed paths
- deleted test doc
- [build] changed check
- add test doc
- fix doc
- fix docs
- [build] fix docs
- fix docs
- fix doc
- fix doc
- fix docs
- fix actions
- revert actions
- fix actions
- test action
- revert test
- Update vcredist checksums (#674) (#675)
- build
- Add tests for editors
- Disable precompiled version on release build
- Add files to gitignore
- Updated max ermitted node.js version
- Changed node.js minor version
- Fix package build (#703)
- Fix check action (#704)
- Add macos builder archive deploy (#706)
- Add vboxtexter to deploy
- Fix build scripts
- libvlc: added build in docker for win32 and win64
- Refactoring build mobile version
- Add module osign
- Fix bug in deploy osign library
- [develop] npm ci in server dir
- libvlc: added build script for linux
- libvlc: build libvlc for linux in docker
- Add method for replace path of text file with utf8 content (python2&3 support)
- libvlc: added build for mac
- Remove open from io
- libvlc: fixed build for mac_arm64
- Update hard-coded version to v7.5.0
- ignoring timestamps on cache loading
- automatic plugins.dat generation on mac
- MacOS package build refactoring (#718)
- Fix windows package build (#719)
- some restyling in libvlc build script
- Refactoring packages deploy (#720)
- [develop] Use npm ci instead of npm i when building web-apps
- Revert partially "[develop] Use npm ci instead of npm i when building web-apps"
- Change deploy with libvlc option
- Fix typo
- Fix typo
- Fix build
- Fix remove repo method
- Refactoring
- Remove gsttools in vlc package
- [develop] Edit readme to mount server volume; Set up debug logging and start test example in docker by default;
- [develop] Modify readme
- [develop] Fix readme(linter)
- [develop] Fix readme(linter)
- [develop] Fix readme(linter)
- [develop] Fix readme(linter)
- [develop] Fix readme(linter)
- [linter] Set code_block_line_length linter rule to 300
- [linter] Fix gitflow
- [linter] Fix gitflow
- [develop] Update readme
- [develop] Fix readme
- [develop] Fix readme
- [develop] Update Dockerfile and instruction
- [develop] Edit readme for linux
- [develop] Fix new supervisor config path
- [develop] Fix Dockerfile
- [develop] Fix Dockerfile
- [develop] Fix linux path
- [develop] Fix missing platform
- [develop] Add check__docker_dependencies call for external sdkjs and web-apps
- [develop] Fix typo
- [develop] Add ALLOW_PRIVATE_IP_ADDRESS (to fix http://localhost/example/images/logo.png insertion)
- [develop] Fix markdown errors
- [develop] Fix markdown errors
- [develop] Add note; remove branch from Dockerfile to prepare for master
- Increased minimum required version of Node.js
- Fix build
- Fix remove repo method
- Add method for deploy dictionaries
- [libvlc] Remove dependency on qtmultimedia.
- [libvlc] Small fix for mac builds
- [libvlc] Fixes for linux and win builds
- Fix closure maps deploy path (#733)
- Add support config addon for each platform
- Fix build
- Add dictiories path to doctrenderer.config
- Fix dictionaries deploy
- Added console matafile tester
- Add patch for xcode 15.0
- Add catch error on file copying
- Add vlc-cache-gen to deploy folder
- Update windows package DesktopEditors vlc plugin cache
- Small fix
- Fix update DesktopEditors vlc plugin cache (#739)
- Fix check config option
- Fix typo
- Fix vlc-cache-gen paths (#740)
- Fix build icu for android on mac
- [develop] Fix Dockerfile. Allow branding command line params
- [develop] Fix markdown errors
- Update windows desktop build (#742)
- Add some plugins by default. drawio, zotero, speech intput.
- Deploy marketplace plugin from base repo
- Fix typo
- Fix copy directory with long names
- Fix del dir with long filenames
- Move speechrecognition to server only
- Fix deploy plugin store in desktop
